kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventListener.java

```diff
@@ -0,0 +1,110 @@
+package org.hypertrace.core.kafka.event.listener;
+
+import com.typesafe.config.Config;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.BiConsumer;
+import lombok.Value;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.serialization.Deserializer;
+
+/**
+ * KafkaEventListener consumes events from a single Kafka Topic and on every message invokes
+ * provided callbacks. The callback invocation is done in a separate thread and needs to be
+ * concurrent safe.
+ */
+public class KafkaEventListener<K, V> implements AutoCloseable {
+  private final KafkaEventListenerThread<K, V> kafkaEventListenerThread;
+
+  private KafkaEventListener(KafkaEventListenerThread<K, V> kafkaEventListenerThread) {
+    this.kafkaEventListenerThread = kafkaEventListenerThread;
+  }
+
+  @Override
+  public void close() throws Exception {
+    kafkaEventListenerThread.interrupt();
+    kafkaEventListenerThread.join(Duration.ofSeconds(10).toMillis());
+  }
+
+  public static final class Builder<K, V> {
+    KafkaEventListenerThreadConfig<K, V> kafkaEventListenerThreadConfig;
+    List<BiConsumer<? super K, ? super V>> callbacks = new ArrayList<>();
+    ExecutorService executorService;
+
+    public Builder<K, V> addKafkaTopic(
+        String consumerName,
+        Config kafkaConfig,
+        Deserializer<K> keyDeserializer,
+        Deserializer<V> valueDeserializer) {
+      assertAbsenceOfKakfaTopic();
+      kafkaEventListenerThreadConfig =
+          new KafkaEventListenerThreadConfig<>(
+              consumerName, kafkaConfig, keyDeserializer, valueDeserializer, null, false);
+      return this;
+    }
+
+    public Builder<K, V> addKafkaConsumer(
```

Comment:
so this and `addKafkaTopic` are two alternative ways of registering consumers right? If so, I'd treat them as overloads and give them the same name. Pull the creation of consumer in the first case into this class, which allows you to clean up the thread config to take just a (name, config, consumer) and similarly clean up the thread.